services/spar/test-integration/Test/Spar/ScimSpec.hs

```diff
@@ -139,61 +144,113 @@ specUsers = describe "operations with users" $ do
     describe "PUT /Users" $ do
         it "responds with 4xx (just making sure...)" $ do
             env <- ask
-            user <- randomSCIMUser
-            (tok, _) <- registerIdPAndSCIMToken
-            putUser_ (Just tok) Nothing user (env ^. teSpar)
+            user <- randomScimUser
+            (tok, _) <- registerIdPAndScimToken
+            updateUser_ (Just tok) Nothing user (env ^. teSpar)
                 !!! assertTrue_ (inRange (400, 499) . statusCode)
 
     describe "PUT /Users/:id" $ do
         it "updates the user attributes in scim_user" $ do
-            env <- ask
-            user <- randomSCIMUser
-            (tok, _) <- registerIdPAndSCIMToken
+            -- Create a user via SCIM
+            user <- randomScimUser
+            (tok, _) <- registerIdPAndScimToken
             storedUser <- createUser tok user
             let userid = scimUserId storedUser
-            user' <- randomSCIMUser
-            putUser_ (Just tok) (Just userid) user' (env ^. teSpar)
-                !!! const 200 === statusCode
-            getresp <- getUser_ (Just tok) userid (env ^. teSpar)
-                <!! const 200 === statusCode
-            let storedUser' :: SCIM.Class.User.StoredUser = decodeBody' getresp
+            -- Overwrite the user with another randomly-generated user
+            user' <- randomScimUser
+            updatedUser <- updateUser tok userid user'
+            -- Get the updated user and check that it matches the user returned
+            -- by 'updateUser'
+            storedUser' <- getUser tok userid
+            liftIO $ updatedUser `shouldBe` storedUser'
+            -- Check that the updated user also matches the data that we sent
+            -- with 'updateUser'
             liftIO $ do
-                SCIM.value (SCIM.thing storedUser) `shouldBe` user
-                SCIM.value (SCIM.thing storedUser') `shouldBe` user'
-                SCIM.id (SCIM.thing storedUser') `shouldBe` SCIM.id (SCIM.thing storedUser)
-                let meta  = SCIM.meta storedUser
-                    meta' = SCIM.meta storedUser'
-                SCIM.resourceType meta `shouldBe` SCIM.resourceType meta'
-                SCIM.created  meta `shouldBe` SCIM.created  meta'
-                SCIM.version meta `shouldBe` SCIM.version meta'
-                SCIM.location meta `shouldBe` SCIM.location meta'
+                Scim.value (Scim.thing storedUser') `shouldBe` user'
+                Scim.id (Scim.thing storedUser') `shouldBe` Scim.id (Scim.thing storedUser)
+                let meta  = Scim.meta storedUser
+                    meta' = Scim.meta storedUser'
+                Scim.version meta `shouldNotBe` Scim.version meta'
+                Scim.resourceType meta `shouldBe` Scim.resourceType meta'
+                Scim.created meta `shouldBe` Scim.created meta'
+                Scim.location meta `shouldBe` Scim.location meta'
+
+        it "works fine when neither name nor handle are changed" $ do
```

Comment:
This test is new. It tests that PUT /Users is not dumb (because I don't trust it not to be dumb).